lib/dynflow/director.rb

```diff
@@ -144,24 +146,26 @@ def terminate
     def unless_done(manager, work_items)
       return [] unless manager
       if manager.done?
-        finish_manager(manager)
-        return []
+        try_to_rescue(manager) || finish_manager(manager)
       else
         return work_items
       end
     end
 
-    def finish_manager(manager)
+    def try_to_rescue(manager)
+      rescue!(manager) if rescue?(manager)
+    end
+
+    def finish_manager(manager, store: true)
+      update_execution_plan_state(manager) if store
+      return []
+    ensure
       @execution_plan_managers.delete(manager.execution_plan.id)
-      if rescue?(manager)
-        rescue!(manager)
-      else
-        set_future(manager)
-      end
+      set_future(manager)
     end
 
     def rescue?(manager)
-      if @world.terminating? || !(@world.auto_rescue && manager.execution_plan.state == :paused)
+      if @world.terminating? || !(@world.auto_rescue && manager.execution_plan.error?)
```

Comment:
We needed to do this, otherwise the plan in paused/pending would get rescued as well